lib/Service/SynchronizationService.php

```diff
@@ -1969,15 +1969,14 @@ private function fetchFile(Source $source, string $endpoint, array $config, stri
             throw new Exception('Could not write file: no filename could be determined');
         }
 
-		// Write file with OpenRegister ObjectService.
 		$objectService = $this->containerInterface->get('OCA\OpenRegister\Service\ObjectService');
-		$file = $objectService->addFile(object: $objectId, fileName: $filename, base64Content: $response['body'], share: isset($config['autoShare']) ? $config['autoShare'] : false);
+		$objectEntity = $objectService->findByUuid(uuid: $objectId);
 
-        // Attach passed down tags
         $tags[] = "object:$objectId";
-        if ($file instanceof File === true && isset($tags) === true && empty($tags) === false) {
-			$this->attachTagsToFile(fileId: $file->getId(), tags: $tags);
-        }
+
+		// Write file with OpenRegister FileService.
+		$fileService = $this->containerInterface->get('OCA\OpenRegister\Service\FileService');
```

Comment:
What if OpenConnector runs without OpenRegister installed. Shouldn't we catch an exception here?